service/worker/handler/consistency_check_handlers.go

```diff
@@ -0,0 +1,272 @@
+	if _, err = s.taskClient.EnqueueContext(ctx, task); err != nil && !errors.Is(err, asynq.ErrDuplicateTask) && !errors.Is(err, asynq.ErrTaskIDConflict) {
+		return fmt.Errorf("unable to enqueue consistency check list task: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) checkConsistencyForObject(ctx context.Context, record *storage.ConsistencyCheckRecord) error {
+	if err := s.storageSvc.AddToConsistencyCheckSet(ctx, record); err != nil {
+		return fmt.Errorf("unable to add storage to consistency check set %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) HandleConsistencyCheckReadiness(ctx context.Context, t *asynq.Task) (err error) {
+	var readinessPayload tasks.ConsistencyCheckReadinessPayload
+	if err := json.Unmarshal(t.Payload(), &readinessPayload); err != nil {
+		return fmt.Errorf("unable to unmarshal paylaod: %w", err)
+	}
+
+	scheduledCounter, err := s.storageSvc.GetConsistencyCheckScheduledCounter(ctx, readinessPayload.ID)
+	if err != nil {
+		return fmt.Errorf("unable to get scheduled counter: %w", err)
+	}
+
+	completedCounter, err := s.storageSvc.GetConsistencyCheckCompletedCounter(ctx, readinessPayload.ID)
+	if err != nil {
+		return fmt.Errorf("unable to get completed counter: %w", err)
+	}
+
+	if scheduledCounter != completedCounter {
+		return &dom.ErrRateLimitExceeded{RetryIn: util.DurationJitter(time.Second, time.Second*2)}
+	}
+
+	if err := s.storageSvc.SetConsistencyCheckReadiness(ctx, readinessPayload.ID, true); err != nil {
+		return fmt.Errorf("unable to set readiness to true: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) HandleConsistencyCheckDelete(ctx context.Context, t *asynq.Task) (err error) {
+	var payload tasks.ConsistencyCheckDeletePayload
+	if err := json.Unmarshal(t.Payload(), &payload); err != nil {
+		return fmt.Errorf("unable to unmarshal paylaod: %w", err)
+	}
+
+	if err := s.storageSvc.DeleteAllConsistencyCheckSets(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete consistency check sets: %w", err)
+	}
+	if err := s.storageSvc.DeleteAllLastListedConsistencyCheckObj(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete last listed objects: %w", err)
+	}
+	if err := s.storageSvc.DeleteConsistencyCheckScheduledCounter(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete scheduled counter: %w", err)
+	}
+	if err := s.storageSvc.DeleteConsistencyCheckCompletedCounter(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete completed counter: %w", err)
+	}
+	if err := s.storageSvc.DeleteConsistencyCheckID(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete consistency check id: %w", err)
+	}
+	if err := s.storageSvc.DeleteConsistencyCheckStorages(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete consistency check storages: %w", err)
+	}
+	if err := s.storageSvc.DeleteConsistencyCheckReadiness(ctx, payload.ID); err != nil {
+		return fmt.Errorf("unable to delete consistency check readiness flag: %w", err)
+	}
```

Comment:
i was not able to find  places where these delete methods are called separately. Is it a good idea to to have a single `ClenaupCheck` mehtod calling all redis commands in batch?

Reply:
Vast majority of these commands are unlinking single key.
`DeleteAllConsistencyCheckSets` and `DeleteAllLastListedConsistencyCheckObj` potentially can unlink many keys.
But the benefit of `unlink` usage is that deletion is not made in real time. 
After `unlink` key becomes undiscoverable. But actual deletion is managed by redis garbage collector.

Reply:
Added pipeline to both functions expecting more than one key.